packages/web/src/apollo/sseLink.tsx

```diff
@@ -0,0 +1,109 @@
+import type { HttpOptions } from '@apollo/client'
+import {
+  ApolloLink,
+  Operation,
+  FetchResult,
+  Observable,
+} from '@apollo/client/core'
+import { print } from 'graphql'
+import { createClient, ClientOptions, Client } from 'graphql-sse'
```

Comment:
This stuff will mean these things will be included in the bundle for users even if they're not using sse (or deploying in a way sse is supported). 

**Suggestion:** 
I recently discovered that our RedwoodApolloProvider takes a linkFactory. I.e. you can append your own links to it. 

```tsx
// App.tsx

import { construcApolloLinkWithSse } from '@redwoodjs/web/apollo/sse'


const App = () => (
  <FatalErrorBoundary page={FatalErrorPage}>
    <RedwoodProvider titleTemplate="%PageTitle | %AppTitle">
      <AuthProvider>
        <RedwoodApolloProvider
          useAuth={useAuth}
          logLevel="debug"
          graphQLClientConfig={{
            link: (rwLinks) => {
              return constructApolloLinkWithSse(rwLinks) 
            },
          }}
        >
```

Reply:
I can see your point, but doesn't feel terribly elegant. 

How far do we go? Do we keep making Providers for SSR and Subscriptions and with or without SSE?

I'd rather get this PR in so I can move forward and get feedback and then tackle:

* auth token optimization
* sse link / provider optimization
* sse with single connection mode and http2 dev support
*

Reply:
In the case of the suspense provider, its so that I don't break main for everyone - at some point the plan is to unify it (neither Apollo client nor I am ready for this just yet - we're still using the experimental nextjs client). Eventually there won't be a separate suspense provider

Yeah not elegant - think it needs more exploration - was just a suggestion to see if it gave you some ideas. We do have to be mindful of including things that aren't required by most people - but if you want to come back to this its up to you! I imagine this is the reason realtime was a separate package.

Reply:
Ok, I did some digging into the link and config and I see your point:

```ts
const App = () => (
  <FatalErrorBoundary page={FatalErrorPage}>
    <RedwoodProvider titleTemplate="%PageTitle | %AppTitle">
      <AuthProvider>
        <RedwoodApolloProvider
          useAuth={useAuth}
          logLevel="debug"
          graphQLClientConfig={{
            link: (rwLinks) => {
              return constructApolloLinkWithSse(rwLinks) 
            },
          }}
        >
```

it would make sense to only have that link if realtime and subscription are enabled. --simiar to have the sse plugin is added only if they are configured.

Maybe the realtime package has a `RedwoodRealtimeApolloProvider` can adds in the terminating link?